packages/ui/src/components/Tooltip/index.tsx

```diff
@@ -164,6 +174,7 @@ const Tooltip = ({
   const unmountTooltip = useCallback(() => {
     setVisibleInDom(false)
     setReverseAnimation(false)
+    timer.current = undefined
```

Comment:
Should we also set debounceTimer.current to undefined ?

Reply:
@ModuloM `unmountTooltip()` function is being called once debounce timer is done:

```tsx
  const unmountTooltip = useCallback(() => {
    setVisibleInDom(false)
    setReverseAnimation(false)
    timer.current = undefined

    window.removeEventListener('scroll', onScrollDetected, true)
  }, [onScrollDetected])

  const onPointerEvent = useCallback(
    (isVisible: boolean) => () => {
      if (!isVisible && tooltipRef.current && !debounceTimer.current) {
        debounceTimer.current = setTimeout(() => { // <== DEBOUNCE TIMEOUT IS SET HERE
          setReverseAnimation(true)
          timer.current = setTimeout(() => unmountTooltip(), ANIMATION_DURATION) // <=== AND UNMOUNT IS CALLED RIGHT HERE
        }, 200)
      } else {
        if (timer.current) {
          setReverseAnimation(false)
          clearTimeout(timer.current)
          timer.current = undefined
        }
        if (debounceTimer.current) {
          clearTimeout(debounceTimer.current)
          debounceTimer.current = undefined
        }
        setVisibleInDom(isVisible)
      }
    },
    [unmountTooltip],
  )
```